include/particle.h

```diff
@@ -262,6 +262,26 @@ class Particle : public ParticleBase<Tdim> {
   //! Assign material id of this particle to nodes
   void append_material_id_to_nodes() const override;
 
+  //! Has neighbour particles
+  bool has_neighbours() const override { return neighbours_.size(); };
+
+  //! Return the number of neighbour particles
+  unsigned nneighbours() const override { return neighbours_.size(); };
+
+  //! Add a neighbour particle
+  //! \param[in] neighbour_id id of the neighbouring particle
+  //! \retval insertion_status Return the successful addition of a node
+  bool add_neighbour(mpm::Index neighbour_id) override;
```

Comment:
Where is `add_neighbour` function used?

Reply:
If you create a new particle, instead of searching the neighbours again, it is better to just add the created particle id into the existing particles. That's true that it's not used, but I think it is useful.

Reply:
Shall we add it when we start using it?

Reply:
We may have to recompute neighbours of all the particles at each step anyway, so I don't think this function helps much. If we see an issue in performance, we can think again. For now, let's remove this function.

Reply:
The `compute_particle_neighbours` with a particular `cell` should improve this performance, without any additional function at the particle level.